samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,67 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %s\n", SDL_IsGameController(gameControllerIndex)?"Yes":"No");
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("\n");
+
+  while (true) {
+    // Fetch current GameController state
+    SDL_GameControllerUpdate();
+    debugClearScreen();
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_A),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_A));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_B),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_B));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_X),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_X));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_Y),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_Y));
+    debugPrint("%s:%i                 ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_BACK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_BACK));
```

Comment:
Why does this one have so many spaces? Wouldn't the amount of padding depend on the length of the string returned by `SDL_GameControllerGetStringForButton`?

Reply:
Yes, that is true. Padding just made the formatting nice for the current output - so far I have not seen any reason why names should change between controller types.

Reply:
- What if you connect a 360 gamepad to the original Xbox?
- What about PC gamepads?
- What about Playstation gamepads?
- What about steering wheels where these might be named differently ("Shift paddle left" / "Shift paddle right") for usability?

I think one of the benefits of the nxdk is that we can support such devices.
So we shouldn't hardcode it.

If we really want to assume that these are constant names, then we don't have to read the names from `SDL_GameControllerGetStringForButton` either. But that defeats the point of using these APIs.

*If* we hardcode the padding, we should at least add a comment. But I'd still consider it bad code, given we have cross-file dependencies about these strings.

Putting each field on a different line, or also padding the names isn't too hard either.

Reply:
Fixed, thanks. :+1:

Reply:
As these names are still constant (SDL2 still lacks labels for gamecontrollers), this isn't too critical after all. The current solution is still better though. Thanks!